src/managers/ChannelManager.js

```diff
@@ -13,7 +13,11 @@ let cacheWarningEmitted = false;
 class ChannelManager extends CachedManager {
   constructor(client, iterable) {
     super(client, Channel, iterable);
-    if (!cacheWarningEmitted && this._cache.constructor.name !== 'Collection') {
+    const defaultCaching =
+      this._cache.constructor.name === 'Collection' ||
```

Comment:
what about `this._cache.constructor === Collection` ?

Reply:
This is an intentional hardcode on the name

It also technically allows for an extension of Collection the way that discord.js used to due it (e.g `Collection extends BaseCollection`) that won't output the error.